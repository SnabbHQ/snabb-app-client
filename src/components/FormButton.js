/**
 * # FormButton.js
 *
 * Display a button that responds to onPress and is colored appropriately
 */
'use strict'
/**
 * ## Imports
 *
 * React
 */
import React from 'react'
import{ StyleSheet, View } from 'react-native'
import {Button} from 'native-base'

class FormButton extends React.Component {
  render () {
    return (
      <View style={styles.signin}>
        <Button style={styles.button}
                textStyle={{fontSize: 18}}
                isDisabled={this.props.isDisabled}
                onPress={this.props.onPress}>
          {this.props.buttonText}
        </Button>
      </View>
    )
  }
}

var styles = StyleSheet.create({
  signin: {
    marginLeft: 10,
    marginRight: 10
  },
  button: {
    backgroundColor: '#FF3366',
    borderColor: '#FF3366'
  }
})

export default FormButton
