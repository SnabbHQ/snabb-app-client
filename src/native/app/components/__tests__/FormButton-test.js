/**
 * # FormButton-test.js
 *
 * This class tests that the form button displays correctly
 *
 * *Note:* if you want to understand the structures here, add a
 * ```console.log``` and then ```npm test```.
 */


/** This is due to the fact that NativeBase does not support yet jest. Mock until it does
 * https://github.com/GeekyAnts/NativeBase/issues/272 **/
jest.mock('Button', () => 'Button');

/**
* ## Imports
 */
import React from 'react';
import FormButton from '../FormButton';
import renderer from 'react/lib/ReactTestRenderer';

it('FormButton', () => {
  const props = {
    isDisabled: false,
    onPress: () => {},
    buttonText: 'TestString',
  };
  const tree = renderer.create(<FormButton {...props} />).toJSON();
  expect(tree).toMatchSnapshot();
});
