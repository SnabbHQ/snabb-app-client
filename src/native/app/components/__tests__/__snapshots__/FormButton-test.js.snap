exports[`test FormButton 1`] = `
<View
  style={
    Object {
      "marginLeft": 10,
      "marginRight": 10
    }
  }>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    onLayout={undefined}
    onResponderGrant={[Function bound touchableHandleResponderGrant]}
    onResponderMove={[Function bound touchableHandleResponderMove]}
    onResponderRelease={[Function bound touchableHandleResponderRelease]}
    onResponderTerminate={[Function bound touchableHandleResponderTerminate]}
    onResponderTerminationRequest={[Function bound touchableHandleResponderTerminationRequest]}
    onStartShouldSetResponder={[Function bound touchableHandleStartShouldSetResponder]}
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FF3366",
        "borderColor": "#FF3366",
        "borderRadius": 8,
        "borderWidth": 1,
        "flexDirection": "row",
        "height": 44,
        "justifyContent": "center",
        "marginBottom": 10,
        "opacity": 1
      }
    }
    testID={undefined}>
    <Text
      allowFontScaling={undefined}
      style={
        Array [
          Object {
            "alignSelf": "center",
            "fontSize": 18
          },
          Object {
            "fontSize": 18
          }
        ]
      }>
      TestString
    </Text>
  </View>
</View>
`;
